crates/wash-cli/src/util.rs

```diff
@@ -32,6 +39,52 @@ pub fn json_str_to_msgpack_bytes(payload: &str) -> Result<Vec<u8>> {
     rmp_serde::to_vec_named(&json).context("failed to encode JSON as msgpack")
 }
 
+/// Ensure that the given plugin directory exists or return the default plugin dir path
+pub async fn ensure_plugin_dir(dir: Option<impl AsRef<Path>>) -> Result<PathBuf> {
+    let plugin_dir = match dir.map(|dir| dir.as_ref().to_path_buf()) {
+        Some(dir) => dir,
+        None => cfg_dir()?.join(PLUGIN_DIR),
+    };
+
+    if !tokio::fs::try_exists(&plugin_dir).await.unwrap_or(false) {
+        tokio::fs::create_dir(&plugin_dir).await?;
+    }
+    Ok(plugin_dir)
+}
+
+/// Helper for loading plugins from the given directory
+pub async fn load_plugins(plugin_dir: impl AsRef<Path>) -> anyhow::Result<SubcommandRunner> {
+    let mut readdir = tokio::fs::read_dir(&plugin_dir)
+        .await
+        .context("Unable to read plugin directory")?;
+
+    let mut plugins = SubcommandRunner::new().context("Could not inititalize plugin runner")?;
+
+    // We load each plugin separately so we only warn if a plugin fails to load
+    while let Some(entry) = readdir.next_entry().await.transpose() {
+        let entry = match entry {
+            Ok(entry) => entry,
+            Err(e) => {
+                eprintln!("WARN: Could not read plugin directory entry. Skipping: {e:?}");
```

Comment:
nit; shouldn't these be `tracing` calls?

Reply:
I was doing them as warnings but I realized by default the logger is set to error level. These aren't fatal but I wanted them to be warnings the user would see so they knew why something wasn't working. @brooksmtownsend thoughts here?